src/adapter/etl-adapter-excel/src/Flow/ETL/Adapter/Excel/RowsNormalizer/ExcelRowsNormalizer.php

```
@@ -74,16 +70,11 @@ private function normalizeEntry(Entry $entry) : bool|float|int|string|null
             DateTimeEntry::class => $entry->value()?->format($this->dateTimeFormat),
             DateEntry::class => $entry->value()?->format($this->dateFormat),
             TimeEntry::class => $entry->value()?->format($this->timeFormat),
-            UuidEntry::class => $entry->toString(),
             EnumEntry::class => $this->normalizeEnumEntry($entry),
             JsonEntry::class,
             ListEntry::class,
             MapEntry::class,
             StructureEntry::class => $this->normalizeToJson($entry->value()),
-            XMLEntry::class,
-            XMLElementEntry::class,
-            HTMLEntry::class,
-            HTMLElementEntry::class => $entry->toString(),
```

I don't think there is anything wrong with explicit conversion, maybe even exception on default would be a good idea in case we introduce some new entry type

---

The problem with new types is that they will be skipped anyway till added explicitly in any normalizer. The only workaround I see is to make a generic normalizer, with some explicit ones if needed for particular adapters. Cause i.e. we have already 
- csv,
- json,
- Parquet
- and now Excel
- potentially we can also have Google Sheets one,

---

they won't if there will be an exception on default case, no?

---

If we fail hardly then yes, but still I think we can consider extracting a generic one ;)

---

I took already at least 3 attempts to solve this problem 😂 The main challenge is that different destinations supports different structures, for example CSV requires nested structures serialized, meanwhile json not, parquet requires them to be kept as array etc 

Don't get me wrong, it would be awesome to at least extract the repetitive parts, but I'm having hard time finding them and making the abstraction easy to use/maintain